Packages/com.unity.inputsystem/InputSystem/Controls/InputControlPath.cs

```diff
@@ -539,7 +539,7 @@ private static bool StringMatches(Substring str, InternedString matchTo)
             // Can't compare lengths here because str may contain wildcards and
             // thus be shorter than matchTo and still match.
 
-            var matchToLowerCase = matchTo.ToLower();
+            var matchToStringUpper = matchTo.ToString().ToUpperInvariant();
```

Comment:
The `ToString()` will cost a GC allocation as it needs to create a new temporary string.

I looked at `matchTo` and why you couldn't just to `ToUpperInvariant` and I found it's because it's an `InternedString`. The docs say this is a special string class designed exactly for what you are doing here, case-insensitive comparisons.

However that class looks like it needs to be updated too as it might also be broken for turkish letters. Maybe it makes sense for it to store internally in uppercase instead and then you wouldn't need `ToString()` here either. I'm not sure how much `InternedString` is used, so it might be quite a bit of work.

Reply:
this is a lot of work. changing InternedString is a big thing because it entails changes at many spots